src/charts/brush-range-editor.js

```diff
@@ -38,6 +38,17 @@ export default function BrushRangeEditor (_container) {
   // events
   const dispatcher = d3.dispatch("rangeChange")
 
+  function handleFocus (selection) {
```

Comment:
nice

Reply:
I'd like to do something similar for a `handleBlur` function but it's getting pretty ugly:

```
  const handleBlur = (cache, config) => (comparator) => () => {
    const domain = scales.xScale.domain()
    const selection = comparator === "max" ? cache.inputMax : cache.inputMin
    const range = comparator === "max" ? cache.rangeMax : cache.rangeMin
    const compareValue = comparator === "max" ? (cache.rangeMin || domain[0]) : (cache.rangeMax || domain[1])
    const oldValue = comparator === "max" ? (cache.rangeMax || domain[1]) : (cache.rangeMax || domain[1])
    const newValue = stringToType(selection.text(), config.keyType)
    const passesBoundsCheck = comparator === "max" ? (newValue > comparatorValue) : (newValue < comparatorValue)

    if (newValue !== compareValue) {
      if (passesBoundsCheck) {
        range = newValue
        const extent = comparator === "max" ? [compareValue, range] : [range, compareValue]
        dispatcher.call(
          "rangeChange",
          this,
          { extent }
        )
      } else {
        const text = oldValue instanceof Date
          ? d3.utcFormat(config.dateFormat)(oldValue)
          : oldValue
        selection.text(text)
      }
    }
  }
```

there's probably a cleaner way of doing it, but perhaps it would still be too abstract?

Reply:
For the record, I also prefer without this handleBlur abstraction, as too much logic has to go in there to handle both cases.